### PR TITLE
Week2-Task1

### PR DESCRIPTION
**Pull request by cht8687**

### CODE REVIEW

File: cis194/week2/robert-chang/src/LogAnalysis.hs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+getIntFromJust s = fromJust . safeIntReader $ s 
+
+isJustValue :: String -> Bool
+isJustValue s = isJust . safeIntReader $ s
+
+parseMessage :: String -> LogMessage
+parseMessage s = case words s of 
+             ("E": level : timestamp : msg)  -> 
+               if (and [isJustValue level, isJustValue timestamp])
+                  then LogMessage (Error $ getIntFromJust $ level) (getIntFromJust timestamp) (unwords msg)
```
Comment (stevemao) [inline cis194/week2/robert-chang/src/LogAnalysis.hs]: `Error $ getIntFromJust $ level` can be `Error . getIntFromJust $ level` I think

File: cis194/week2/robert-chang/src/LogAnalysis.hs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+getIntFromJust :: String -> Int 
+getIntFromJust s = fromJust . safeIntReader $ s 
+
+isJustValue :: String -> Bool
+isJustValue s = isJust . safeIntReader $ s
+
+parseMessage :: String -> LogMessage
+parseMessage s = case words s of 
+             ("E": level : timestamp : msg)  ->
```
Comment (shineli1984) [inline cis194/week2/robert-chang/src/LogAnalysis.hs]: nit: We usually have 4 space indentation and 2 space for `where` clause
```Haskell
parseMessage s = case words s of 
    ("E" ....
```

File: cis194/week2/robert-chang/src/LogAnalysis.hs (hunk 1 above)
Comment (shineli1984) [inline cis194/week2/robert-chang/src/LogAnalysis.hs]: Nit: align `if`, `then` and `else`
```Haskell
if ....
then ....
else ....
```

Also, when we check if something is `Just ...` then use `fromJust`, we can always use pattern match instead.
You can have nested pattern match. Or an advance trick is to use view patterns extension: https://gitlab.haskell.org/ghc/ghc/wikis/view-patterns

File: cis194/week2/robert-chang/src/LogAnalysis.hs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                  else Unknown s
+             ("I": timestamp : msg) ->
+               if (isJustValue timestamp)
+                  then LogMessage Info (getIntFromJust timestamp) (unwords msg)
+                  else Unknown s
+             ("W": timestamp : msg) ->
+               if (isJustValue timestamp)
+                  then LogMessage Warning (getIntFromJust timestamp) (unwords msg)
+                  else Unknown s
+             __ -> Unknown s
```
Comment (shineli1984) [inline cis194/week2/robert-chang/src/LogAnalysis.hs]: nit: The convention is a single `_`.

File: cis194/week2/robert-chang/src/LogAnalysis.hs (hunk 2 above)
Comment (cht8687) [reply cis194/week2/robert-chang/src/LogAnalysis.hs]: thanks!

File: cis194/week2/robert-chang/src/LogAnalysis.hs (hunk 1 above)
Comment (cht8687) [reply cis194/week2/robert-chang/src/LogAnalysis.hs]: This is mind blowing...

**Pull request merged**